src/fsharp/NameResolution.fs

```diff
@@ -770,57 +788,376 @@ let AddUnionCases2 bulkAddMode (eUnqualifiedItems: UnqualifiedItems) (ucrefs: Un
             let item = Item.UnionCase(GeneralizeUnionCaseRef ucref, false)
             acc.Add (ucref.CaseName, item))
 
-let AddStaticContentOfTyconRefToNameEnv (g:TcGlobals) (amap: Import.ImportMap) ad m (nenv: NameResolutionEnv) (tcref:TyconRef) =
-    // If OpenStaticClasses is not enabled then don't do this
-    if amap.g.langVersion.SupportsFeature LanguageFeature.OpenStaticClasses then
-        let ty = generalizedTyconRef tcref
-        let infoReader = InfoReader(g,amap)
-        let items =
-            [| let methGroups = 
-                   AllMethInfosOfTypeInScope ResultCollectionSettings.AllResults infoReader nenv None ad PreferOverrides m ty
-                   |> List.groupBy (fun m -> m.LogicalName)
-
-               for (methName, methGroup) in methGroups do
-                   let methGroup = methGroup |> List.filter (fun m -> not m.IsInstance && not m.IsClassConstructor)
-                   if not methGroup.IsEmpty then
-                       yield KeyValuePair(methName, Item.MethodGroup(methName, methGroup, None)) 
-           
-               let propInfos = 
-                   AllPropInfosOfTypeInScope ResultCollectionSettings.AllResults infoReader nenv None ad PreferOverrides m ty
-                   |> List.groupBy (fun m -> m.PropertyName)
-               
-               for (propName, propInfos) in propInfos do
-                   let propInfos = propInfos |> List.filter (fun m -> m.IsStatic)
-                   for propInfo in propInfos do 
-                       yield KeyValuePair(propName , Item.Property(propName,[propInfo]))
-            
-               let fields =
-                  infoReader.GetILFieldInfosOfType(None, ad, m, ty)
-                  |> List.groupBy (fun f -> f.FieldName)
-
-               for (fieldName, fieldInfos) in fields do
-                   let fieldInfos = fieldInfos |> List.filter (fun fi -> fi.IsStatic)
-                   for fieldInfo in fieldInfos do
-                       yield KeyValuePair(fieldName, Item.ILField(fieldInfo))
-             |]
-
-        { nenv with eUnqualifiedItems = nenv.eUnqualifiedItems.AddAndMarkAsCollapsible items }
-    else
-        nenv
+//-------------------------------------------------------------------------
+// TypeNameResolutionInfo
+//-------------------------------------------------------------------------
+
+/// Indicates whether we are resolving type names to type definitions or to constructor methods.
+type TypeNameResolutionFlag =
```

Comment:
Note for this and others - I understand this was just moved up in the file from its previous definition. Might still be good to mark as struct though since it's just a flag

Reply:
Not my code, I only moved them to access them in places that are closer to the top.